security-analysis/security-analysis-api/src/main/java/com/powsybl/security/interceptors/SecurityAnalysisInterceptor.java

```diff
@@ -15,9 +18,80 @@
  */
 public interface SecurityAnalysisInterceptor {
 
+    /**
+     * @Deprected. Use {@link #onPreContingencyResult(LimitViolationsResult, SecurityAnalysisResultContext)}
+     * Callback after the pre-contingency analysis result is created
+     * @param context The running context
+     * @param preContingencyResult
+     */
+    @Deprecated
     void onPreContingencyResult(RunningContext context, LimitViolationsResult preContingencyResult);
 
-    void onPostContingencyResult(RunningContext context, PostContingencyResult postContingencyResult);
+    /**
+     * @Deprecated. Use {@link #onPostContingencyResult(PostContingencyResult, SecurityAnalysisResultContext)}
+     * @param context
+     * @param postContingencyResult
+     */
+    @Deprecated
+    default void onPostContingencyResult(RunningContext context, PostContingencyResult postContingencyResult) {
+        onPostContingencyResult(postContingencyResult, context);
+    }
+
+    /**
+     * @Deprecated. Use {@link #onSecurityAnalysisResult(SecurityAnalysisResult, SecurityAnalysisResultContext)}
+     * Callback after the result is created
+     * @param context
+     * @param result
+     */
+    @Deprecated
+    default void onSecurityAnalysisResult(RunningContext context, SecurityAnalysisResult result) {
+        onSecurityAnalysisResult(result, context);
+    }
+
+    /**
+     * Callback after the pre-contingency result is built.
+     * @param preContingencyResult
+     * @param context
+     */
+    default void onPreContingencyResult(LimitViolationsResult preContingencyResult, SecurityAnalysisResultContext context) {
+        onPreContingencyResult(new RunningContext(context.getNetwork(), VariantManagerConstants.INITIAL_VARIANT_ID), preContingencyResult);
```

Comment:
I finally don't think it's a good idea to add those default implementations, because they will not have the same behaviour as the original methods (because passed variant ID is not the same).
I think it's less error-prone to force the implementations to adapt on migration.